common/Util-desktop.cpp

```diff
@@ -160,6 +162,56 @@ std::size_t getTotalSystemMemoryKb()
     return totalMemKb;
 }
 
+size_t getMaxConcurrentTCPConnections(const size_t minimum)
+{
+#ifdef __linux__
+    auto readDecimal = [](const char* path, const size_t defaultValue) -> size_t {
+        const int fd = ::open(path, O_RDONLY);
+        if (fd<0)
+            return defaultValue;
+        char data[1024+1]; // includes EOS
+        ::memset(data, 0, sizeof(data));
+        size_t consumed = 0;
+        while (sizeof(data)-consumed > 1 /* ex-EOS */)
+        {
+            ssize_t n;
+            while ((n = ::read(fd, data+consumed, sizeof(data)-consumed-1 /* ex-EOS */)) < 0 && errno == EINTR) {}
+            if (n < 0) // Error
+            {
+                consumed = 0;
+                break;
+            }
+            else if (n == 0) // EOF
+                break;
+            consumed += size_t(n);
+        }
+        ::close(fd);
+        if (consumed==0)
+            return defaultValue;
+        char* endptr = nullptr;
+        errno = 0;  // Flush previous error indicator. Reminder: errno is thread-local
+        const unsigned long long num = std::strtoull(data, &endptr, 10);
+        if (0 != errno || nullptr == endptr || endptr == data || num > std::numeric_limits<size_t>::max())
+            return defaultValue;
+        return size_t(num);
+    };
+    // - 4 tcp_max_orphans sockets per MB
+    // - {4096M -> 16384}, {16384M -> 65536}, {65407M -> 262144}, ...
+    const ssize_t tcp_max_orphans = readDecimal("/proc/sys/net/ipv4/tcp_max_orphans", 0); // ignored if n/a
+    const ssize_t nf_conntrack_max = readDecimal("/proc/sys/net/nf_conntrack_max", 0); // ignored if n/a
+    const size_t res1 = nf_conntrack_max > 0 ? std::min(tcp_max_orphans, nf_conntrack_max) : tcp_max_orphans;
```

Comment:
res1 and res2 ? reminds me of the very un-loved c1, c2, c3 etc.

Reply:
res1 -> res_raw (res = result)
res2 -> res_clipped
OK